File: gitdot-web/packages/api/src/endpoint/repository/update-repository-settings.ts
import { z } from "zod";

import { RepositorySettingsResource } from "../../resource";
import { CommitFilterResource } from "../../resource/settings";
import type { Endpoint } from "../endpoint";

export const UpdateRepositorySettingsRequest = z.object({
  commit_filters: z.array(CommitFilterResource).optional(),
});
export type UpdateRepositorySettingsRequest = z.infer<
  typeof UpdateRepositorySettingsRequest
>;

export const UpdateRepositorySettingsResponse = RepositorySettingsResource;
export type UpdateRepositorySettingsResponse = z.infer<
  typeof UpdateRepositorySettingsResponse
>;

export const UpdateRepositorySettings = {
  path: "/repository/{owner}/{repo}/settings",
  method: "PATCH",
  request: UpdateRepositorySettingsRequest,
  response: UpdateRepositorySettingsResponse,
} as const satisfies Endpoint;
export type UpdateRepositorySettings = typeof UpdateRepositorySettings;

File: gitdot-api/src/endpoint/repository/update_repository_settings.rs
use serde::{Deserialize, Serialize};

use crate::{
    endpoint::Endpoint,
    resource::{repository::RepositorySettingsResource, settings::CommitFilterResource},
};

pub struct UpdateRepositorySettings;

impl Endpoint for UpdateRepositorySettings {
    const PATH: &'static str = "/repository/{owner}/{repo}/settings";
    const METHOD: http::Method = http::Method::PATCH;

    type Request = UpdateRepositorySettingsRequest;
    type Response = UpdateRepositorySettingsResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct UpdateRepositorySettingsRequest {
    pub commit_filters: Option<Vec<CommitFilterResource>>,
}

pub type UpdateRepositorySettingsResponse = RepositorySettingsResource;
